src/main/kotlin/com/dansplugins/factionsystem/locks/MfLockService.kt

```diff
@@ -43,6 +55,64 @@ class MfLockService(private val plugin: MedievalFactions, private val repository
         ServiceFailure(exception.toServiceFailureType(), "Service error: ${exception.message}", exception)
     }
 
+    public fun unlock(player: Player, block: Block) {
```

Comment:
Perhaps remove the player parameter and have this return some kind of result object with a failure message (either Result4k or its own type) - that would make this reusable beyond contexts which involve a Bukkit Player, and keeps view-related code out of the service.
If you use Result4k it also gives you the nice pattern of:
```kotlin
.onFailure { return it }
```
which lets you bubble up failures from methods you call

Reply:
The player parameter has been removed and this method now returns a status indicator.